File: aha_integration/src/executors/test_api.ts
/**
 * Test script: fetches features from Aha! API and prints those assigned to Frederik Chettouh.
 */

import "dotenv/config";
import { AhaApiClient } from "../api/client.js";

const ASSIGNEE_MATCH = "Frederik Chettouh";

interface AssignedToUser {
  id?: string;
  name?: string;
  email?: string;
}

interface Feature {
  reference_num: string;
  name: string;
  assigned_to_user?: AssignedToUser | null;
}

interface FeaturesResponse {
  features: Feature[];
  pagination?: { total_pages: number; current_page: number };
}

async function main() {
  const apiKey = process.env.AHA_API_KEY;
  if (!apiKey) {
    console.error("Missing AHA_API_KEY in .env");
    process.exit(1);
  }

  const client = new AhaApiClient({ apiKey });
  const allFeatures: Feature[] = [];
  let page = 1;
  let totalPages = 1;

  do {
    const res = await client.get<FeaturesResponse>("/features", {
      fields: "name,reference_num,assigned_to_user",
      page: String(page),
      per_page: "100",
    });

    allFeatures.push(...(res.features ?? []));
    totalPages = res.pagination?.total_pages ?? 1;
    page++;
  } while (page <= totalPages);

  const matchesAssignee = (f: Feature): boolean => {
    const assignee = f.assigned_to_user;
    if (!assignee) return false;
    const name = typeof assignee === "string" ? assignee : assignee.name ?? "";
    const email = typeof assignee === "string" ? "" : assignee.email ?? "";
    return (
      name.toLowerCase().includes(ASSIGNEE_MATCH.toLowerCase()) ||
      email.toLowerCase().includes(ASSIGNEE_MATCH.toLowerCase())
    );
  };

  const assigned = allFeatures.filter(matchesAssignee);

  console.log("\n--- Features assigned to Frederik Chettouh ---\n");
  if (assigned.length === 0) {
    console.log("(none found)\n");
    return;
  }
  assigned.forEach((f) => {
    console.log(`  ${f.reference_num}  ${f.name}`);
  });
  console.log(`\nTotal: ${assigned.length} feature(s)\n`);
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});

File: aha_integration/src/executors/test_api.py
#!/usr/bin/env python3
"""
Test script: fetches features from Aha! API and prints those assigned to Frederik Chettouh.
"""

import os
import sys

_project_root = os.path.abspath(os.path.join(os.path.dirname(__file__), "..", ".."))
sys.path.insert(0, os.path.join(_project_root, "src"))
os.chdir(_project_root)

from dotenv import load_dotenv

load_dotenv()

from api.client import AhaApiClient

ASSIGNEE_MATCH = "Frederik Chettouh"


def matches_assignee(assigned_to_user) -> bool:
    if not assigned_to_user:
        return False
    if isinstance(assigned_to_user, str):
        return ASSIGNEE_MATCH.lower() in assigned_to_user.lower()
    name = (assigned_to_user.get("name") or "").lower()
    email = (assigned_to_user.get("email") or "").lower()
    return (
        ASSIGNEE_MATCH.lower() in name
        or ASSIGNEE_MATCH.lower() in email
    )


def main():
    api_key = os.getenv("AHA_API_KEY")
    if not api_key:
        print("Missing AHA_API_KEY in .env", file=sys.stderr)
        sys.exit(1)

    client = AhaApiClient(api_key)
    all_features = []
    page = 1
    total_pages = 1

    while page <= total_pages:
        res = client.get(
            "/features",
            params={
                "fields": "name,reference_num,assigned_to_user",
                "page": str(page),
                "per_page": "100",
            },
        )
        features = res.get("features", [])
        all_features.extend(features)
        pagination = res.get("pagination", {})
        total_pages = pagination.get("total_pages", 1)
        page += 1

    assigned = [f for f in all_features if matches_assignee(f.get("assigned_to_user"))]

    print("\n--- Features assigned to Frederik Chettouh ---\n")
    if not assigned:
        print("(none found)\n")
        return
    for f in assigned:
        print(f"  {f.get('reference_num', '?')}  {f.get('name', '')}")
    print(f"\nTotal: {len(assigned)} feature(s)\n")


if __name__ == "__main__":
    main()
